cou/steps/backup.py

```diff
@@ -78,7 +77,7 @@ def get_database_app(model_name: Optional[str] = None) -> Optional[str]:
     :returns: Name of the mysql-innodb-cluster application name
     :rtype: str
     """
-    candidates = get_upgrade_candidates(model_name=model_name)
+    candidates = await get_upgrade_candidates(model_name=model_name)
```

Comment:
Analysis object currently just contains openstack components.

Reply:
then we'd be better off not throwing away the mysql app when doing the analysis and save the extra juju query

Reply:
I do not think it is a good idea to pollute structure just for this. I've made some refactoring, removed upgrade_utils and use the cached version of juju_model. Since it has previously called by analysis part. It will be definitely in cache.

Reply:
I think it's a valid idea using the Analysis/Application objects to create a plan for backup, but this can be refactored in the future.

I've removed the OpenStack filtering in the analyze module because we might want to use non-openstack charms like rabbit, mysql and etc.